scripts/src/public/routes/users.ts

```diff
@@ -219,3 +224,49 @@ export const logoutUser = async function(req: AuthenticatedRequest, res: Respons
 	await logoutService(req.context.user, req.context.token);
 	res.status(204).send();
 } as any as RequestHandler;
+
+export type BulkUserCreationRequest = Request & { body: { app_id: string, user_data: Array<[string, string]> } };
+
+export const bulkUserCreation = async function(req: BulkUserCreationRequest, res: Response) {
+	const jwtList = req.body.user_data as Array<[string, string]>;
+	const firstJwt = jsonwebtoken.decode(jwtList[0][0], { complete: true }) as JWTContent<Partial<RegisterPayload>, "register">; // Use the app id from the first JWT
+	const app_id = firstJwt.payload.iss;
+	logger().info(`bulkUserCreation for ${ app_id }`);
+	const app = await Application.get(app_id);
+	if (!app) {
+		throw NoSuchApp(app_id);
+	}
+	let allowedCreations: number = app.config.bulk_user_creation_allowed || 0;
+	if (!allowedCreations || allowedCreations < jwtList.length) {
+		throw BulkUserCreation(app.id, jwtList.length, allowedCreations);
+	}
+	await app.save();
+	res.write(`requestId: ${ httpContext.get("reqId") }\n`);
+	await batch(jwtList, app.config.limits.minute_registration, 61 * 1000, async (sublist: Array<[string, string]>, firstIndexOfChunk) => {
+			await Promise.all(sublist.map(async ([jwt, publicAddress], index) => {
+				const currentItem = firstIndexOfChunk + (index + 1);
```

Comment:
This calculation seems very confusing.
I think using a simple for loop and calling a function on a slice of users would look much more readable than using a callback chain.

Something like:
```
for (i = 0; i < jwt.length; i += BATCH) {
  slice = jwt.slice(i, i+BATCH);
  registerBatch(slice);
}
```